### PR TITLE
feat(EMI-1595): add text and tooltip to saves list header

### PR DESCRIPTION
**Pull request by leamotta**
The type of this PR is: **Feat**

This PR solves [EMI-1595]

### Description

Inform users about the Partner Offer new project by showing a toast on the saves page.

### Mobile:
Screenshot 2024-01-10 at 16 47 47

### Web:
Screenshot 2024-01-10 at 16 48 03

#### Note: 
I found it hard to follow the designs perfectly. Tried having the ProgressiveOnborading only on the specific link but it generated issues in the responsive view, as it created a div that I couldn't wrap around and instead the text was displayed as 2 columns like so
Screenshot 2024-01-11 at 14 59 32

[EMI-1595]: https://artsyproduct.atlassian.net/browse/EMI-1595?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by narikazu — APPROVED** [review]
LGTM

File: src/Components/ProgressiveOnboarding/ProgressiveOnboardingSaveTitle.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  return (
+    <ProgressiveOnboardingPopover
+      name={ALERT_ID}
+      placement="top-end"
+      onClose={handleClose}
+      popover={
+        <Text variant="xs">
+          <>{t("collectorSaves.artworkListsHeader.curateYourList")}</>
```
Comment (rquartararo) [inline src/Components/ProgressiveOnboarding/ProgressiveOnboardingSaveTitle.tsx:39]: Super minor, but I don't think the react fragment is needed.

**Review by MrSltun — APPROVED** [review]
Looks good! 🌟 Thank you Leandro! 🦅

**Review by rquartararo — APPROVED** [review]
LGTM!

File: src/Apps/CollectorProfile/Routes/Saves/Components/ArtworkListsHeader.tsx [hunk 2]
```diff
@@ -57,7 +64,19 @@ export const ArtworkListsHeader: FC<ArtworkListsHeaderProps> = ({
...
         >
           <Text variant="sm-display" color="black60">
-            {t("collectorSaves.artworkListsHeader.curateYourList")}
+            {isPartnerOfferEnabled ? (
+              <ProgressiveOnboardingSaveTitle>
+                {t("collectorSaves.artworkListsHeader.curateYourList") +
+                  " " +
+                  t("collectorSaves.artworkListsHeader.connector") +
+                  " "}
+                <RouterLink to="TODO.com">
```
Comment (iskounen) [inline src/Apps/CollectorProfile/Routes/Saves/Components/ArtworkListsHeader.tsx:73]: Is there a Jira ticket to update this so that we don't forget?

Comment (leamotta) [reply src/Apps/CollectorProfile/Routes/Saves/Components/ArtworkListsHeader.tsx:73]: I added a question in this ticket so see if we can get it in before merge, I'll wait until tomorrow

Comment (iskounen) [reply src/Apps/CollectorProfile/Routes/Saves/Components/ArtworkListsHeader.tsx:73]: Got it. Thank you.

Comment (MrSltun) [reply src/Apps/CollectorProfile/Routes/Saves/Components/ArtworkListsHeader.tsx:73]: I already created a ticket for it [EMI-1644](https://artsyproduct.atlassian.net/browse/EMI-1644) in case it will take a while
